closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -106,34 +98,52 @@ public String[] run() {
                     }
                 });
 
-        boolean isOsSupported, isArchSupported;
+        boolean isOsSupported;
+        boolean isArchSupported;
+
         // Check whether the NSS FIPS solution is supported.
-        isOsSupported = false;
-        for (String os: supportedPlatformsNSS.get("OS")) {
-            if (props[2].contains(os)) {
-                isOsSupported = true;
-            }
-        }
-        isArchSupported = false;
-        for (String arch: supportedPlatformsNSS.get("Arch")) {
-            if (props[3].contains(arch)) {
-                isArchSupported = true;
-            }
+        switch (props[2]) { // os.name
+        case "Linux":
+            isOsSupported = true;
+            break;
+        default:
+            isOsSupported = false;
+            break;
+        }
+
+        switch (props[3]) { // os.arch
+        case "amd64":
+        case "ppc64le":
+        case "s390x":
+            isArchSupported = true;
+            break;
+        default:
+            isArchSupported = false;
+            break;
         }
         isNSSSupported = isOsSupported && isArchSupported;
 
         // Check whether the OpenJCEPlus FIPS solution is supported.
-        isOsSupported = false;
-        for (String os: supportedPlatformsOpenJCEPlus.get("OS")) {
-            if (props[2].contains(os)) {
-                isOsSupported = true;
-            }
-        }
-        isArchSupported = false;
-        for (String arch: supportedPlatformsOpenJCEPlus.get("Arch")) {
-            if (props[3].contains(arch)) {
-                isArchSupported = true;
-            }
+        switch (props[2]) { // os.name
+        case "AIX":
+        case "Linux":
+        case "Windows":
```

Comment:
Only the non-fips OpenJCEPlus provider is supported on aarch64/x86-64 Mac, so no need to add the change there.

Reply:
How does this accomplish the title of this pull request ("Enable aarch64/x86-64 Mac bundling OpenJCEPlus") without recognizing "Mac OS X" anywhere in this class?

Reply:
1. RestrictSecurity class is currently only utilized for FIPS use cases on the fips-related providers, e,g, NSS and OpenJCEPlusFIPS, it does not address the non-fips such as OpenJCEPlus provider at the moment. Only the non-fips OpenJCEPlus provider is supported on aarch64/x86-64 Mac, so I dont think we need to make any changes to the OS and aarch.
2. Updated the name of `isOpenJCEPlusSupported` to `isOpenJCEPlusFIPSSupported`, think it could be more clear.
3. The building procedure is included in the Lib.gmk file, and the native building part is in the Copy.gmk file.

Reply:
Please explain how the changes in `RestrictedSecurity.java` relate to the subject of this pull request. To me they seem unrelated and are more about replacing maps with switch statements to check for supported architecture and O/S combinations, in which case changes to that file belong in a separate change focused on that refactoring (and perhaps removing use of `doPrivileged()`).

Reply:
> I can open another PR to fix those if you want.

Yes, please do.

Reply:
Reverted the changes and will open another PR for this doPrivileged issue.